lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -1722,21 +1722,28 @@ bool PullbackCloner::Implementation::run() {
   }
 
   // Create pullback blocks and arguments, visiting original blocks in
-  // post-order post-dominance order.
-  SmallVector<SILBasicBlock *, 8> postOrderPostDomOrder;
-  // Start from the root node, which may have a marker `nullptr` block if
-  // there are multiple roots.
-  PostOrderPostDominanceOrder postDomOrder(postDomInfo->getRootNode(),
-                                           postOrderInfo, original.size());
-  while (auto *origNode = postDomOrder.getNext()) {
-    auto *origBB = origNode->getBlock();
-    postDomOrder.pushChildren(origNode);
-    // If node is the `nullptr` marker basic block, do not push it.
-    if (!origBB)
-      continue;
-    postOrderPostDomOrder.push_back(origBB);
+  // dominance order.
```

Comment:
I don't think ["dominance order"](https://en.wikipedia.org/wiki/Dominator_(graph_theory)) is an accurate description for the new traversal order. Dominance order traversal starts at the function entry basic block and continues to successors based on dominance information.

"BFS from the original exit block" seems like an exactly accurate description. How about:

```suggestion
  // Create pullback blocks and arguments, visiting original blocks using BFS
  // starting from the original exit block. Unvisited original basic blocks
  // (e.g unreachable blocks) are not relevant for pullback generation and thus
  // ignored.
```